app/src/main/java/ch/eureka/eurekapp/ui/notes/SelfNotesViewModel.kt

```diff
@@ -0,0 +1,239 @@
+                    errorMsg = displayedError,
+                    currentMessage = currentMessage,
+                    isSending = isSending,
+                    isCloudStorageEnabled = isCloud)
+              }
+          .stateIn(
+              scope = viewModelScope,
+              started = SharingStarted.WhileSubscribed(5000),
+              initialValue = SelfNotesUIState(isLoading = true))
+
+  init {
+    // Monitor Connectivity to trigger Sync
+    viewModelScope.launch {
+      try {
+        // Safe call in case ConnectivityObserver isn't initialized
+        ConnectivityObserverProvider.connectivityObserver.isConnected.collect { isOnline ->
+          if (isOnline) {
+            val syncedCount = repository.syncPendingNotes()
+            if (syncedCount > 0) {
+              _errorMsg.value = "Back online: Uploaded $syncedCount notes"
+            }
+          }
+        }
+      } catch (e: Exception) {
+        Log.e("SelfNotesViewModel", "Connectivity Observer error", e)
+      }
+    }
+  }
+
+  /**
+   * Toggles the storage mode between Local-only and Cloud.
+   *
+   * When enabling Cloud mode, this triggers an immediate sync of existing local notes. Updates the
+   * [_errorMsg] flow with a status message upon completion.
+   *
+   * @param enableCloud True to enable Cloud storage/sync, False for Local-only.
+   */
+  fun toggleStorageMode(enableCloud: Boolean) {
+    viewModelScope.launch {
+      val syncedCount = repository.setStorageMode(enableCloud)
+      if (enableCloud) {
+        if (syncedCount > 0) {
+          _errorMsg.value = "Switched to Cloud: Synced $syncedCount notes"
+        } else {
+          _errorMsg.value = "Switched to Cloud Storage"
+        }
+      } else {
+        _errorMsg.value = "Switched to Local Storage (Private)"
+      }
+    }
+  }
+
+  /**
+   * Updates the text of the message currently being composed.
+   *
+   * @param text The new text entered by the user.
+   */
+  fun updateMessage(text: String) {
+    _currentMessage.value = text
+  }
+
+  /**
+   * Creates and saves a new note.
+   *
+   * The note is created with the current input text. It is saved via the repository, which handles
+   * local storage and potential cloud upload. If successful, clears the input and schedules a
+   * background worker for redundancy.
+   */
+  fun sendNote() {
+    val currentMessage = _currentMessage.value.trim()
+    if (currentMessage.isEmpty()) return
+
+    _isSending.value = true
+
+    viewModelScope.launch(dispatcher) {
+      val message =
+          Message(
+              messageID = IdGenerator.generateMessageId(),
+              text = currentMessage,
+              createdAt = Timestamp.now())
+
+      repository
+          .createNote(message)
+          .fold(
+              onSuccess = {
+                _currentMessage.value = ""
+                _isSending.value = false
+                _errorMsg.value = null
+                scheduleWorker()
+              },
```

Comment:
You're scheduling the sync worker every time a note is created, even when cloud storage is disabled. I suggest to check the state first:

```suggestion
onSuccess = {
    _currentMessage.value = ""
    _isSending.value = false
    _errorMsg.value = null
    if (uiState.value.isCloudStorageEnabled) {
      scheduleWorker()
    }
  }
```